src/lib/Security/EventListener/JWT/AuthenticationSuccessSubscriber.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Rest\Security\EventListener\JWT;
+
+use Ibexa\Contracts\Core\Repository\PermissionResolver;
+use Ibexa\Core\MVC\Symfony\Security\UserInterface as IbexaUser;
+use Lexik\Bundle\JWTAuthenticationBundle\Event\AuthenticationSuccessEvent;
+use Lexik\Bundle\JWTAuthenticationBundle\Events;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+final readonly class AuthenticationSuccessSubscriber implements EventSubscriberInterface
+{
+    public function __construct(private PermissionResolver $permissionResolver)
+    {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            Events::AUTHENTICATION_SUCCESS => ['onAuthenticationSuccess', 10],
+        ];
+    }
+
+    public function onAuthenticationSuccess(AuthenticationSuccessEvent $event): void
+    {
+        $user = $event->getUser();
+        if ($user instanceof IbexaUser) {
+            $this->permissionResolver->setCurrentUserReference($user->getAPIUser());
+        }
+
+        $this->normalizeResponseToRest($event);
+    }
+
+    /*
+     * This method provides BC compatibility for the JWT Token REST response
+     * since the new Lexik/JWT json_login authenticator changes its form.
+     *
+     * @deprecated: Drop on releasing the new REST API version.
+     */
+    private function normalizeResponseToRest(AuthenticationSuccessEvent $event): void
+    {
+        $token = $event->getData()['token'];
```

Comment:
I would personally add checks against the presence of this value in data, and return early if it's not there.

It's an event system, so it can be changed by the end-developer by using `Event::setData()`, just like we do.

Reply:
I will add some exception in that case and cover that in the dedicated test case.

Reply:
Added via https://github.com/ibexa/rest/pull/101/commits/93b52904a5ce8c8c9876390e073a0d830e6b5ffd.